denizen_scripts/survival/repo-link/items/system/soul_item_system.dsc

```diff
@@ -308,86 +418,115 @@ item_system_build_item:
-      - define "name:<[rarity_color]><[name1].replace[_].with[<&sp>].to_titlecase> of the <[item].nbt[active_soul].to_titlecase>"
+      - define name "<[rarity_color]><[name1].replace[_].with[<&sp>].to_titlecase> of the <[item].nbt[active_soul].to_titlecase>"
     - else:
-      - define "name:<[rarity_color]><[name1].replace[_].with[<&sp>].to_titlecase>"
-    #- define lore:|:<script[item_system_global_data].yaml_key[settings.lore.top].parse[parsed]>
-    - define lore:|:<script[item_system_global_data].yaml_key[settings.lore.middle.global].parse[parsed]>
-    - if <[item].material.name.matches[<script[item_system_global_data].yaml_key[regex_type_check.weapon]>]>:
-      - define damage <script[item_system_global_data].yaml_key[defaults.damage.<[item].material.name>]>
-      - define lore:|:<script[item_system_global_data].yaml_key[settings.lore.middle.weapons].parsed>
-    - if <[item].material.name.matches[<script[item_system_global_data].yaml_key[regex_type_check.armor]>]>:
-      - define armor <script[item_system_global_data].yaml_key[defaults.armor.<[item].material.name>]>
-      - define lore:|:<script[item_system_global_data].yaml_key[settings.lore.middle.armors].parsed>
-    ##- if <[item].nbt[buffs].unescaped.as_list||<[item].nbt[soul_buffs].unescaped.as_list||null>> != null:
-    ##  - define lore:|:<&a>Buffs
-    - foreach <[item].nbt[buffs].unescaped.as_list||<[item].nbt[soul_buffs].unescaped.as_list||<list[]>>> as:buff:
-      - if <[buff]> == none:
-        - foreach next
-      - define final_value <[buff].after[,]>
-      - define lore:|:<script[item_system_global_data].yaml_key[settings.lore.middle.buffs.<[buff].before[,]>].parsed>
-    - foreach <[item].nbt[debuffs].unescaped.as_list||<[item].nbt[soul_debuffs].unescaped.as_list||<list[]>>> as:debuff:
-      - if <[debuff]> == none:
-        - foreach next
-      - define final_value <[debuff].after[,]>
-      - define lore:|:<script[item_system_global_data].yaml_key[settings.lore.middle.debuffs.<[debuff].before[,]>].parsed>
+      - define name <[rarity_color]><[name1].replace[_].with[<&sp>].to_titlecase>
+
+
+    - define lore <list.include[<script[item_system_global_data].parsed_key[settings.lore.top]>]>
+    - define lore <[lore].include[<script[item_system_global_data].parsed_key[settings.lore.middle.global]>]>
+    - if <[item].material.name.matches[<script[item_system_global_data].data_key[regex_type_check.weapon]>]>:
+      - define damage <script[item_system_global_data].data_key[defaults.damage.<[item].material.name>]>
+      - define lore <[Lore].include[<script[item_system_global_data].parsed_key[settings.lore.middle.weapons]>]>
+
+    - if <[item].material.name.matches[<script[item_system_global_data].data_key[regex_type_check.armor]>]>:
+      - define armor <script[item_system_global_data].data_key[defaults.armor.<[item].material.name>]>
+      - define lore <[Lore].include[<script[item_system_global_data].parsed_key[settings.lore.middle.armors]>]>
```

Comment:
I was going to `foreach` the list of lore available until I realized there were raw definitions being parsed within the data keys, we ideally could move that directly into the script or consider dynamic logic that would correctly reflect both loops for armor / weapons

Ideally, we should really remove regex, considering we have both:
- A full list of all material names that we could check against the `item_system_global_data.nbt_slots` keys
- Tags that can run this without `ElementTag.matches[]` such as `ElementTag.ends_with[]`